# --- ROCK-COPYRIGHT-NOTE-BEGIN ---
# 
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# Please add additional copyright information _after_ the line containing
# the ROCK-COPYRIGHT-NOTE-END tag. Otherwise it might get removed by
# the ./scripts/Create-CopyPatch script. Do not edit this copyright text!
# 
# ROCK Linux: rock-src/package/avm/btscanner/get-read.patch
# ROCK Linux is Copyright (C) 1998 - 2006 Clifford Wolf
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# 
# --- ROCK-COPYRIGHT-NOTE-END ---

diff -ruN btscanner-1.0/bs_info.c btscanner-1.0-new/bs_info.c
--- btscanner-1.0/bs_info.c	2003-10-14 18:31:10.000000000 +0200
+++ btscanner-1.0-new/bs_info.c	2005-06-09 11:05:00.000000000 +0200
@@ -160,7 +160,7 @@
 	struct hci_conn_info_req *cr;
 	struct hci_request rq;
 	read_rssi_rp rp_rssi;
-	get_link_quality_rp rp_lq;
+	read_link_quality_rp rp_lq;
 	int err = 0;
 	int leave = 0;;
 
@@ -292,11 +292,11 @@
 		memset(&rq, 0, sizeof(rq));
 		memset(&rp_lq, 0, sizeof(rp_lq));
 		rq.ogf = OGF_STATUS_PARAM;
-		rq.ocf = OCF_GET_LINK_QUALITY;
+		rq.ocf = OCF_READ_LINK_QUALITY;
 		rq.cparam = &handle;
 		rq.clen = 2;
 		rq.rparam = &rp_lq;
-		rq.rlen = GET_LINK_QUALITY_RP_SIZE;
+		rq.rlen = READ_LINK_QUALITY_RP_SIZE;
 
 		if (hci_send_req(dd, &rq, 100) < 0) {
 			p->lq_status = (uint8_t)-1;
